### PR TITLE
use one CrateClient per URL

### PR DESCRIPTION
**Pull request by dobe**

### CODE REVIEW

File: src/main/java/io/crate/client/jdbc/CrateDriver.java [hunk 1]
```diff
@@ -78,6 +126,22 @@ public Connection connect(String url, Properties info) throws SQLException {
         return connection;
     }
 
+
+    private ClientHandle getHandle(String url) {
+        ClientHandle handle;
+        synchronized (clientHandles) {
```
Comment (mfussenegger) [inline src/main/java/io/crate/client/jdbc/CrateDriver.java]: couldn't this be changed to use get/putIfAbsent to avoid synchronized?

Comment (dobe) [inline src/main/java/io/crate/client/jdbc/CrateDriver.java]: the incref needs to be synchronized

File: src/main/java/io/crate/client/jdbc/CrateDriver.java [hunk 2]
```diff
@@ -42,9 +45,56 @@
...
+        public String url() {
+            return url;
+        }
+
+        public void connectionClosed() {
+            synchronized (clientHandles) {
+                if (--refCount == 0) {
+                    clientHandles.remove(this.url);
+                }
+                ;
```
Comment (mfussenegger) [inline src/main/java/io/crate/client/jdbc/CrateDriver.java]: remove semicolon

**Comment by mfussenegger** [comment]
otherwise ack

**Pull request merged**